src/hooks/Tokens.ts

```diff
@@ -57,9 +59,53 @@ export function useAllTokens(): { [address: string]: Token } {
   return useTokensFromMap(allTokens, true)
 }
 
+interface BridgeInfo {
+  [chainId: SupportedChainId | number | string]: {
+    tokenAddress: string
+    originBridgeAddress: string
+    destBridgeAddress: string
+  }
+}
 export function useUnsupportedTokens(): { [address: string]: Token } {
+  const { chainId } = useActiveWeb3React()
+  const listsByUrl = useAppSelector((state) => state.lists.byUrl)
   const unsupportedTokensMap = useUnsupportedTokenList()
-  return useTokensFromMap(unsupportedTokensMap, false)
+  const unsupportedTokens = useTokensFromMap(unsupportedTokensMap, false)
+
+  // checks the default L2 lists to see if `bridgeInfo` has an L1 address value that is unsupported
+  const L2InferredBlockedTokens: { [address: string]: Token } = useMemo(() => {
+    if (!chainId || ![SupportedChainId.ARBITRUM_ONE, SupportedChainId.OPTIMISM].includes(chainId)) {
```

Comment:
Can this be written more generically? Maybe have the `constants/lists` file export an L2_CHAIN_IDS array? I'm afraid it might be missed in the future if more L2 chains are added.

Reply:
We do that currently, but that array of chainIds includes L2 testnets where the token lists needed here aren't active by default. I'll take a look at options for exporting constants with and without the testnet ids.